csharp/test/Ice/idleTimeout/Client.cs

```diff
@@ -8,6 +8,9 @@ public override async Task runAsync(string[] args)
     {
         var properties = createTestProperties(ref args);
         properties.setProperty("Ice.Connection.IdleTimeout", "1");
+
+        // TODO: temporary work-around for IceSSL always sending messages asynchronously.
```

Comment:
Otherwise, testIdleCheckDoesNotAbortConnectionWhenThreadPoolIsExhausted fails (unexpected idle check on the client-side).

Note that testIdleCheckDoesNotAbortConnectionWhenThreadPoolIsExhausted goal is to test that the server does not abort the connection. It's "by accident" that this test also highlighted this issue with thread pool starvation (heartbeat send gets delayed with async writes and IceSSL C# uses only async writes).

Reply:
I don't think it's specific to IceSSL / C#. We shouldn't assume that writes can still go on when the Ice thread pool is starved.

Reply:
In practice, when the thread pool is starved, sync writes succeed while async writes wait (since no thread is available to send them). I agree we can't/shouldn't count on all writes to be sync.